src/tracing-policy.ts

```diff
@@ -42,51 +39,72 @@ export class RateLimiterPolicy implements TracePolicy {
   }
 }
 
-export class FilterPolicy implements TracePolicy {
-  constructor(
-      private basePolicy: TracePolicy,
-      private filterUrls: Array<string|RegExp>) {}
+class URLFilter implements TracePolicyPredicate<string> {
+  constructor(private readonly filterUrls: Array<string|RegExp>) {}
 
-  private matches(url: string) {
-    return this.filterUrls.some((candidate) => {
+  shouldTrace(url: string) {
+    return !this.filterUrls.some((candidate) => {
       return (typeof candidate === 'string' && candidate === url) ||
           !!url.match(candidate);
     });
   }
-
-  shouldTrace(dateMillis: number, url: string) {
-    return !this.matches(url) && this.basePolicy.shouldTrace(dateMillis, url);
-  }
 }
 
-export class TraceAllPolicy implements TracePolicy {
-  shouldTrace() {
-    return true;
-  }
+/**
+ * Options for constructing a TracePolicy instance.
+ */
+export interface TracePolicyConfig {
+  /**
+   * A field that controls time-based sampling.
+   */
+  samplingRate: number;
+  /**
+   * A field that controls a url-based filter.
+   */
+  ignoreUrls: Array<string|RegExp>;
 }
 
-export class TraceNonePolicy implements TracePolicy {
-  shouldTrace() {
-    return false;
+/**
+ * A class that makes decisions about whether a trace should be created.
+ */
+export class TracePolicy {
```

Comment:
Should this implement `TracePolicyPredicate<RateAndUrl>`?

Reply:
I opted not to do so just because there's no need to export `TracePolicyPredicate`. The only reason the interface exists is because I would run into issues typing `sampler` as a `Sampler`.

Thanks for looking at this!